pandas/core/common.py

```diff
@@ -1482,10 +1482,11 @@ def _get_dtype_from_object(dtype):
 
         try:
             return _get_dtype_from_object(getattr(np,dtype))
```

Comment:
This would still fail the `dtype='integer'` case, and I'm not sure the latter would work at all without more processing. It's the use case I'm solving with the test.

What case are you trying to solve with the above?